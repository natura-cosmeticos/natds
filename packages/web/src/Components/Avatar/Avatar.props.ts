import { AvatarProps as MaterialAvatarProps } from '@material-ui/core/Avatar/Avatar'
import { IAvatarSizes } from '@naturacosmeticos/natds-styles'

export type AvatarColor = 'primary' | 'secondary' | 'default' | string;
export type AvatarSize = keyof IAvatarSizes;

/**
 * @deprecated `rounded` and `square` variants are deprecated since version 0.19.
 * They will be removed at version 1.0, when all Avatar variants will be `circle` by default.
 */
export type AvatarVariant = 'circle' | 'rounded' | 'square';

export interface IAvatarProps extends Omit<MaterialAvatarProps, 'variant'> {

  /**
   * The color of the Avatar, using the tokens from the current theme
   *
   * @default "default"
   * @optional
   * @type AvatarColors
   */
  color?: AvatarColor;

  /**
   * The possible sizes of the Avatar
   *
   * @default "standard"
   * @optional
   * @todo Adopt new avatar sizes for 1.0
   * @type AvatarSizes
   */
  size?: AvatarSize;

  /**
   * The shape of the avatar.
   *
   * @deprecated `variant` property is deprecated since 0.19.0.
   * It will be removed on 1.0 and all Avatar variants will be `circle` by default.
   */
  variant?: AvatarVariant;

}
