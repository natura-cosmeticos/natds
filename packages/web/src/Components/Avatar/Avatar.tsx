import * as React from 'react'
import MaterialAvatar from '@material-ui/core/Avatar'
import makeStyles from '@material-ui/core/styles/makeStyles'
import { IAvatarProps } from './Avatar.props'
import { IThemeWeb } from '../../Themes'
import styles from './Avatar.styles'
import { Icon } from '../Icon'

/**
 * ## Importing
 *
 * For more advanced usages, check [Material UI Avatar docs](https://material-ui.com/components/avatars/)
 *
 * ```
 * import { Avatar } from "@naturacosmeticos/natds-web";
 * ```
 */
export const Avatar = React.forwardRef<HTMLDivElement, IAvatarProps>((
  props: IAvatarProps, ref
) => {
  const {
    color,
    size,

    /**
     * @deprecated `variant` property is deprecated since 0.19.0.
     * It will be removed on 1.0 and all Avatar variants will be `circle` by default.
     */
    variant,
    ...otherProps
  } = props

  const useStyles = makeStyles(
    (theme: IThemeWeb) => styles({ color, size, theme }), { name: 'NatDsAvatar' }
  )

  const classes = useStyles()

  return (
    <MaterialAvatar
      classes={classes}
      variant={variant}
      {...otherProps}
      ref={ref}
    >
      {props.children || <Icon name="outlined-social-person" />}
    </MaterialAvatar>
  )
})

Avatar.displayName = 'Avatar'
export default Avatar
