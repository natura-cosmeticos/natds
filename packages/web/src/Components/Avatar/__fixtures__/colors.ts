import { AvatarColor } from '../Avatar.props'

export const colors: Record<AvatarColor, AvatarColor> = {
  primary: 'primary',
  secondary: 'secondary',
  default: 'default'
}

export default colors
