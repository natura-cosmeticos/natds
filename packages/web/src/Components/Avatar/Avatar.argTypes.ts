import { colors } from './__fixtures__/colors'
import { sizes } from './__fixtures__/sizes'
import { IAvatarProps } from './Avatar.props'

export const argTypes : Record<keyof IAvatarProps | string, unknown> = {
  alt: {
    control: {
      type: 'text'
    }
  },
  children: {
    control: {
      type: 'object'
    }
  },
  color: {
    control: {
      options: colors,
      type: 'inline-radio'
    },
    defaultValue: colors.default
  },
  component: {
    control: {
      type: 'text'
    },
    description: 'The component used for the root node. Either a `string` to use a DOM element or a component.',
    table: {
      defaultValue: {
        summary: '"div"'
      },
      type: {
        summary: 'elementType'
      }
    }
  },
  imgProps: {
    control: {
      type: 'object'
    }
  },
  size: {
    control: {
      options: sizes,
      type: 'select'
    },
    defaultValue: sizes.standard
  },
  src: {
    control: {
      type: 'text'
    }
  },
  srcSet: {
    control: {
      type: 'text'
    }
  },
  sizes: {
    control: {
      type: 'text'
    }
  },

  /**
   * @deprecated `variant` property is deprecated since version 0.19.
   * It will be removed at 1.0. All Avatar variations will be internally `circle` by default.
   */
  variant: {
    control: {
      options: ['circle'],
      type: 'inline-radio'
    },
    table: {
      defaultValue: {
        summary: '"circle"'
      }
    }
  }
}

export default argTypes
