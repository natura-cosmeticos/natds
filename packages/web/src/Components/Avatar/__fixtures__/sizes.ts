import { AvatarSize } from '../Avatar.props'

export const sizes : Record<AvatarSize, AvatarSize> = {
  tiny: 'tiny',
  small: 'small',
  standard: 'standard',
  large: 'large',
  huge: 'huge'
}

export default sizes
