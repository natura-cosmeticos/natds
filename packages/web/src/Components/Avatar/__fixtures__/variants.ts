import { AvatarVariant } from '../Avatar.props'

/**
 * @deprecated variant is deprecated since 0.19.
 * It will be removed at version 1.0, when all Avatar variants will be `circle` by default.
 */
export const variants : Record<AvatarVariant, AvatarVariant> = {
  circle: 'circle',

  /**
   * @deprecated `rounded` variant is deprecated since 0.19.
   * It will be removed at version 1.0, when all Avatar variants will be `circle` by default.
   */
  rounded: 'rounded',

  /**
   * @deprecated `square` variant is deprecated since 0.19.
   * It will be removed at version 1.0, when all Avatar variants will be `circle` by default.
   */
  square: 'square'
}
