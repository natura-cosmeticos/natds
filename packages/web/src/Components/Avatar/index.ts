import Avatar from './Avatar'

export { IAvatarProps } from './Avatar.props'
export { Avatar }
export default Avatar
