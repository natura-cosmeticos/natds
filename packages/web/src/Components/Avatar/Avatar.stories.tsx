import * as React from 'react'
import { Story } from '@storybook/react'
import { Avatar as component } from './Avatar'
import argTypes from './Avatar.argTypes'

import colors from './__fixtures__/colors'
import sizes from './__fixtures__/sizes'

import { IAvatarProps } from './Avatar.props'

const natNaturaImage = 'https://pbs.twimg.com/profile_images/1308179129267781632/wLDnEiOy_400x400.jpg'
const Avatar : React.ForwardRefExoticComponent<IAvatarProps> = React.lazy(() => import('./Avatar'))

Avatar.displayName = 'Avatar'

export default {
  argTypes,
  component,
  title: 'Components/Avatar'
}

const Template : Story<IAvatarProps> = (args) => <Avatar {...args} />

export const Playground : Story<IAvatarProps> = Template.bind({})
Playground.args = {
  alt: 'Nat Natura holding a cat',
  color: colors.primary,
  src: natNaturaImage,
  size: sizes.huge
}

export const Image : Story<IAvatarProps> = Template.bind({})
Image.args = {
  alt: 'Nat Natura holding a cat',
  src: natNaturaImage,
  size: sizes.huge
}

export const AnonymousWithDefaultFallbackIcon : Story<IAvatarProps> = Template.bind({})
AnonymousWithDefaultFallbackIcon.args = {
  color: colors.primary,
  size: sizes.huge
}

export const Letter : Story<IAvatarProps> = Template.bind({})
Letter.args = {
  children: 'T',
  color: colors.primary,
  size: sizes.huge
}

export const PrimaryColor : Story<IAvatarProps> = Template.bind({})
PrimaryColor.args = {
  color: colors.primary,
  children: 'P',
  size: sizes.standard
}

export const SecondaryColor : Story<IAvatarProps> = Template.bind({})
SecondaryColor.args = {
  color: colors.secondary,
  children: 'S',
  size: sizes.standard
}

export const DefaultColor : Story<IAvatarProps> = Template.bind({})
DefaultColor.args = {
  color: colors.default,
  children: 'D',
  size: sizes.standard
}

export const TinySize : Story<IAvatarProps> = Template.bind({})
TinySize.args = {
  ...Image.args,
  size: sizes.tiny
}

export const SmallSize : Story<IAvatarProps> = Template.bind({})
SmallSize.args = {
  ...Image.args,
  size: sizes.small
}

export const StandardSize : Story<IAvatarProps> = Template.bind({})
StandardSize.args = {
  ...Image.args,
  size: sizes.standard
}

export const LargeSize : Story<IAvatarProps> = Template.bind({})
LargeSize.args = {
  ...Image.args,
  size: sizes.large
}

export const HugeSize : Story<IAvatarProps> = Template.bind({})
HugeSize.args = {
  ...Image.args,
  size: sizes.huge
}
