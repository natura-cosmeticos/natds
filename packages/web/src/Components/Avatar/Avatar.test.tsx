/* eslint-disable @typescript-eslint/ban-ts-comment */
import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'

import Avatar from './Avatar'
import { Provider } from '../../Provider'
import { IThemeWeb, themes } from '../../Themes'
import { colors } from './__fixtures__/colors'
import { variants } from './__fixtures__/variants'
import { AvatarColor, AvatarSize, AvatarVariant } from './Avatar.props'
import { sizes } from './__fixtures__/sizes'

describe('Avatar component', () => {
  const label = 'A'
  let testRenderer : TestRenderer.ReactTestRenderer = {} as TestRenderer.ReactTestRenderer

  describe('given no theme was provided', () => {
    // @ts-ignore
    describe.each(Object.keys(colors))('when color is %p', (color: AvatarColor) => {
      it(`should render ${color} color correctly`, () => {
        testRenderer = TestRenderer.create(<Avatar color={color}>{label}</Avatar>)
        expect(testRenderer).toMatchSnapshot()
      })
    })

    // @ts-ignore
    describe.each(Object.keys(variants))('when variant is %p', (variant: AvatarVariant) => {
      it(`should render ${variant} variant correctly`, () => {
        testRenderer = TestRenderer.create(<Avatar variant={variant}>{label}</Avatar>)
      })
    })

    describe('when initials are provided as children', () => {
      it('should render correctly', () => {
        testRenderer = TestRenderer.create(<Avatar>AB</Avatar>)
        expect(testRenderer).toMatchSnapshot()
      })
    })

    describe('when an icon is provided as children', () => {
      it('should render correctly', () => {
        const MockedIcon = () => <i />

        testRenderer = TestRenderer.create(<Avatar><MockedIcon /></Avatar>)
        expect(testRenderer).toMatchSnapshot()
      })
    })

    describe('when an image is provided as children', () => {
      it('should render correctly', () => {
        const image = 'source/image.png'

        testRenderer = TestRenderer.create(<Avatar alt="" src={image} />)
        expect(testRenderer).toMatchSnapshot()
      })
    })
  })

  describe('given a default theme was provided', () => {
    describe('when no props are overridden', () => {
      it('should render correctly', () => {
        testRenderer = TestRenderer.create(<Provider><Avatar /></Provider>)
        expect(testRenderer).toMatchSnapshot()
      })
    })
  })

  describe('given a specific theme was explicitly provided', () => {
    const theme : IThemeWeb = themes.natura.light

    describe('when no property is provided', () => {
      it('should render correctly', () => {
        testRenderer = TestRenderer.create(<Provider theme={theme}><Avatar /></Provider>)
        expect(testRenderer).toMatchSnapshot()
      })
    })

    // @ts-ignore
    describe.each(Object.keys(sizes))('when size is %p', (size: AvatarSize) => {
      it(`should render ${size} correctly`, () => {
        testRenderer = TestRenderer.create(<Provider theme={theme}><Avatar size={size} /></Provider>)
        expect(testRenderer).toMatchSnapshot()
      })
    })

    // @ts-ignore
    describe.each(Object.keys(colors))('when color is %p', (color: AvatarColor) => {
      it(`should render ${color} correctly`, () => {
        testRenderer = TestRenderer.create(<Provider theme={theme}><Avatar color={color} /></Provider>)
        expect(testRenderer).toMatchSnapshot()
      })
    })
  })
})
