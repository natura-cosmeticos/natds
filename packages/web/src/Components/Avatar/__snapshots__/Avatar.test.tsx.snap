// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component given a default theme was provided when no props are overridden should render correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when color is "default" should render default correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when color is "primary" should render primary correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when color is "secondary" should render secondary correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when no property is provided should render correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when size is "huge" should render huge correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when size is "large" should render large correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when size is "small" should render small correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when size is "standard" should render standard correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given a specific theme was explicitly provided when size is "tiny" should render tiny correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i
    aria-hidden={true}
    className="material-icons MuiIcon-root natds-icons natds-icons-outlined-social-person NatDSIcon-root-3 NatDSIcon-root-4"
  />
</div>
`;

exports[`Avatar component given no theme was provided when an icon is provided as children should render correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-15 MuiAvatar-circular MuiAvatar-colorDefault"
>
  <i />
</div>
`;

exports[`Avatar component given no theme was provided when an image is provided as children should render correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-17 MuiAvatar-circular"
>
  <img
    alt=""
    className="MuiAvatar-img"
    src="source/image.png"
  />
</div>
`;

exports[`Avatar component given no theme was provided when color is "default" should render default color correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-5 MuiAvatar-circular MuiAvatar-colorDefault"
>
  A
</div>
`;

exports[`Avatar component given no theme was provided when color is "primary" should render primary color correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
>
  A
</div>
`;

exports[`Avatar component given no theme was provided when color is "secondary" should render secondary color correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-3 MuiAvatar-circular MuiAvatar-colorDefault"
>
  A
</div>
`;

exports[`Avatar component given no theme was provided when initials are provided as children should render correctly 1`] = `
<div
  className="MuiAvatar-root NatDsAvatar-root-13 MuiAvatar-circular MuiAvatar-colorDefault"
>
  AB
</div>
`;
